fdbserver/SimulatedCluster.actor.cpp

```diff
@@ -2052,6 +2064,19 @@ void setupSimulatedSystem(std::vector<Future<Void>>* systemActors,
 
 	simconfig.db.tenantMode = tenantMode;
 	simconfig.db.encryptionAtRestMode = EncryptionAtRestMode::DISABLED;
+	if (!testConfig.encryptModes.empty()) {
+		simconfig.db.encryptionAtRestMode =
+		    EncryptionAtRestMode::fromString(deterministicRandom()->randomChoice(testConfig.encryptModes));
+	} else if (!testConfig.disableEncryption && deterministicRandom()->coinflip()) {
+		if (tenantMode == TenantMode::DISABLED || tenantMode == TenantMode::OPTIONAL_TENANT ||
+		    deterministicRandom()->coinflip()) {
+			// optional and disabled tenant modes currently only support cluster aware encryption
+			simconfig.db.encryptionAtRestMode = EncryptionAtRestMode::CLUSTER_AWARE;
+		} else {
+			simconfig.db.encryptionAtRestMode = EncryptionAtRestMode::DOMAIN_AWARE;
+		}
+	}
+	TraceEvent("SimulatedClusterEncryptionMode").detail("Mode", simconfig.db.encryptionAtRestMode.toString());
```

Comment:
nit: we can record `TenantMode` as well

Reply:
TenantMode is being recorded by the `SimulatedClusterTenantMode` prior to this